core/src/main/scala/org/apache/spark/deploy/mesos/MesosExternalShuffleService.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.mesos
+
+import java.util.concurrent.CountDownLatch
+
+import scala.collection.mutable
+
+import org.apache.spark.deploy.ExternalShuffleService
+import org.apache.spark.rpc.{RpcAddress, RpcEnv, ThreadSafeRpcEndpoint}
+import org.apache.spark.util.Utils
+import org.apache.spark.{Logging, SecurityManager, SparkConf}
+
+private[mesos] class MesosExternalShuffleService(conf: SparkConf, securityManager: SecurityManager)
+  extends ExternalShuffleService(conf, securityManager) {
+
+  private var rpcEnv: RpcEnv = _
+
+  override def start(): Unit = {
+    super.start()
+    val port = conf.getInt("spark.mesos.shuffle.service.port", 7327)
```

Comment:
One thing you could do is have a `MesosExternalShuffleBlockHandler` that extends `ExternalShuffleBlockHandler`. It would override the `receive` method to also accept a `RegisterDriver` message. Note that because it extends `RpcHandler` there is also a `connectionTerminated` method, so we can try using that one instead.